app/models/user.rb

```diff
@@ -71,7 +74,14 @@ class User < ApplicationRecord
   validates :username,
             presence: true,
             uniqueness: { case_sensitive: false },
-            length: { minimum: 3 }
+            length: { minimum: 3, maximum: 50 },
+            format: { with: USERNAME_REGEXP }
```

Comment:
Let user know which characters are allowed - in error message, but perhaps also in form label. We could replace `Username (not your email address!)` with `Username (letters, numbers and dots allowed)`. I think it'll be automatically clear that it's not an email address.